aips/aip-x.md

```diff
@@ -0,0 +1,707 @@
+---
+aip: (this is determined by the AIP Manager, leave it empty when drafting)
+title: Sign in with Aptos
+author: aptqpv
+Status: Draft # <Draft | Last Call | Accepted | Final | Rejected>
+type: Standard (Ecosystem)
+created: 12/16/2024
+requires: 62
+---
+
+# AIP-X - Sign in with Aptos
+
+## Summary
+
+The "Sign in with Aptos" (SIWA) standard introduces a secure and user-friendly way for users to authenticate to off-chain resources by proving ownership of their blockchain addresses. It simplifies the authentication process by replacing the traditional `connect` + `signMessage` flow in the wallet standard with a streamlined one-click `signIn` method. SIWA leverages decentralized identities to avoid reliance on traditional schemes like SSO while incorporating security measures to combat phishing attacks and improve user visibility.
+
+### Out of scope
+
+The standard relies on public key signature verification. This AIP will not cover specifications or correctness of signature verification.
+
+## High-level Overview
+
+**Decentralized authentication** is a privacy-first solution that empowers individuals to safeguard their information, personalize their experiences, and maintain freedom from censorship.
+
+By authenticating through a wallet, users have full control over their authentication journey. They can decide _what to share_ and _how to log in,_ ensuring a more secure and customizable experience.
+
+For example, a user may choose to use one of the wallet experiences to authenticate into an application:
+
+- **Petra** – _Passwords/Biometric_: - Fully non-custodial. Requires no personal information, ensuring complete privacy and resistance to censorship.
+- **Aptos Connect** – _Social Providers_: Fully non-custodial. Shares limited personal information like name and email but may be subject to censorship by social providers.
+
+> 💡 **_Decentralized authentication transforms wallets into decentralized identities._**
+
+However, while this approach offers significant benefits, it also presents several challenges:
+
+1. **Lack of standardization:** Applications on Aptos lack a unified authentication message format, leading to inconsistencies that confuse users.
+2. **Unreadable messages:** Authentication messages are displayed in plaintext, making them difficult for users to comprehend.
+3. **Limited wallet support:** Without standardization, wallets cannot effectively support authentication UIs.
+4. **Security risks:** Malicious websites can deceive users into signing messages intended for legitimate dApps.
```

Comment:
Isn't the `signMessage` flow domain-separated? Doesn't it include the requesting dapp's URL in the signed message? I guess it is optional?

Reply:
Can you point via URL to AIP-62's `*MessageInput` structs with the optional `application` field, for extra clarity.